### ISSUE TITLE
Implement `column!()` builtin macro

### ISSUE THREAD

**Issue by CohenArthur** [Rust-GCC/gccrs#979]
`column!()` is part of the list of builtin macros enumerated [here](#927)

Its description is as follows:
> Expands to the column number at which it was invoked.

You need to fetch the line number associated to the macro invocation:

```fish
arthur@platypus ~/G/gccrs (add-assert-macro)> cat test.rs
fn main() {
    let c0 = column!();
    let c1 =                             column!();

    println!("cols: {} and {}", c0, c1);
}
arthur@platypus ~/G/gccrs (add-assert-macro)> rustc test.rs
arthur@platypus ~/G/gccrs (add-assert-macro)> ./test
cols: 14 and 42
```

The macro should be replaced by a singular node containing an integer. Each macro builtin is invoked with the macro invocation location, like so
```c++
AST::Fragment
MacroBuiltin::assert(Location invoc_locus /* <- here */, AST::MacroInvocData &invoc);
```
You should be able to dig around the location and find the column number. You can take a look at `MacroBuiltin::file`, which implements the `file!()` macro, for a similar operation.

**Comment by mvvsmk** [comment]
hey @CohenArthur could you assign this issue to me, I would to like to work on it. I have never worked with marcos before and I think this would be a good place to start.

**Comment by tschwinge** [comment]
Fixed via #1004 as far as I can tell.  (Please re-open if anything's missing still.)
